sygnal/apnspushkin.py

```diff
@@ -283,6 +283,15 @@ async def _dispatch_notification_unlimited(
         with self.sygnal.tracer.start_span(
             "apns_dispatch", tags=span_tags, child_of=context.opentracing_span
         ) as span_parent:
+            # Before we build the payload, check that the default_payload is not
+            # malformed and reject the pushkey if it is
+            if device.data:
+                default_payload = device.data.get("default_payload", {})
+                if not isinstance(default_payload, dict):
+                    log.error(
+                        "Default_payload is malformed, this value must be a dict."
```

Comment:
I'd recommend not capitalising the `d` in `default` here, since `default_payload` is a (case sensitive) key.

Reply:
ah that makes sense and I never would have thought of that, thanks for pointing out.